src/effects/lv2/LV2Ports.h

```diff
@@ -83,6 +83,14 @@ struct LV2AtomPortState final {
       , mBuffer{ safenew uint8_t[mpPort->mMinimumSize] }
    {
       assert(mpPort);
+      /*
+       There is no complementary munlock anywhere in the library!
+       Are we mis-using a class meant for plugin implementation?
+       Do we make a resource leak of physical memory pages?
+       Should we just skip this call?
+       Or is mlock a good idea we should implement in our own RingBuffer,
+       which we can also munlock in its destructor?
+       */
```

Comment:
I think the lock is released when `zix_ring_free` is called on a buffer.

Reply:
And docs say that we must lock for RT processing


<html>
<body>
<!--StartFragment--><h2 class="memtitle" style="transition: text-shadow 0.5s linear 0s; margin-right: 15px; padding: 8px; border-top: 1px solid rgb(168, 184, 217); border-left: 1px solid rgb(168, 184, 217); border-right: 1px solid rgb(168, 184, 217); border-top-right-radius: 4px; border-top-left-radius: 4px; margin-bottom: -1px; background-image: url(&quot;nav_f.png&quot;); background-repeat: repeat-x; background-color: rgb(226, 232, 242); line-height: 1.25; font-weight: 300; float: left; color: rgb(0, 0, 0); font-family: Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">zix_ring_mlock()</h2><div class="memitem" style="font: 400 14px / 22px Roboto, sans-serif; padding: 0px; margin-bottom: 10px; margin-right: 5px; transition: box-shadow 0.5s linear 0s; width: 2523px; display: table !important; color: rgb(0, 0, 0); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="memproto" style="font: bold 14px / 22px Roboto, sans-serif; border-top: 1px solid rgb(168, 184, 217); border-left: 1px solid rgb(168, 184, 217); border-right: 1px solid rgb(168, 184, 217); padding: 6px 0px; color: rgb(37, 53, 85); text-shadow: rgba(255, 255, 255, 0.9) 0px 1px 1px; background-color: rgb(223, 229, 241); box-shadow: rgba(0, 0, 0, 0.15) 5px 5px 5px; border-top-right-radius: 4px;">

void zix_ring_mlock | ( | ZixRing * | ring | ) |  
-- | -- | -- | -- | -- | --


</div><div class="memdoc" style="font: 400 14px / 22px Roboto, sans-serif; border-width: 0px 1px 1px; border-bottom-style: solid; border-bottom-color: rgb(168, 184, 217); border-left-style: solid; border-left-color: rgb(168, 184, 217); border-right-style: solid; border-right-color: rgb(168, 184, 217); padding: 6px 10px 2px; background-image: url(&quot;nav_g.png&quot;); background-repeat: repeat-x; background-color: rgb(255, 255, 255); border-bottom-left-radius: 4px; border-bottom-right-radius: 4px; box-shadow: rgba(0, 0, 0, 0.15) 5px 5px 5px;"><p style="font: 400 14px / 22px Roboto, sans-serif;">Lock the ring data into physical memory.</p><p style="font: 400 14px / 22px Roboto, sans-serif;">This function is NOT thread safe or real-time safe, but it should be called after<span> </span><a class="el" href="https://doxy.audacityteam.org/group__zix.html#ga5baff9d9e029c5ba2dfbf116d0d40ae8" style="color: rgb(61, 87, 140); font-weight: bold; text-decoration: none;">zix_ring_new()</a><span> </span>to lock all ring memory to avoid page faults while using the ring (i.e. this function MUST be called first in order for the ring to be truly real-time safe).</p></div></div><!--EndFragment-->
</body>
</html>zix_ring_mlock()
void zix_ring_mlock	(	[ZixRing](https://doxy.audacityteam.org/group__zix.html#gae3e2c7fcac3fd13a91edda72c7176fcd) * 	ring	)	
Lock the ring data into physical memory.

This function is NOT thread safe or real-time safe, but it should be called after [zix_ring_new()](https://doxy.audacityteam.org/group__zix.html#ga5baff9d9e029c5ba2dfbf116d0d40ae8) to lock all ring memory to avoid page faults while using the ring (i.e. this function MUST be called first in order for the ring to be truly real-time safe).

Reply:
> I think the lock is released when `zix_ring_free` is called on a buffer. I don't think that you have to unlock memory before freeing

`man mlock` makes no mention of an interaction with `free`.  And `zix_ring_free` is simply

```
void
zix_ring_free(ZixRing* ring)
{
	free(ring->buf);
	free(ring);
}
```

So I think there can be a leak of physical pages.